roles/setup_dbt2_driver/tasks/main.yml

```diff
@@ -8,11 +8,35 @@
     tasks_from: dbt2_install_packages_driver
 
 - name: Include DBT-2 user creation
-  include_role:
-    name: setup_dbt2
-    tasks_from: dbt2_setup_user
+  include_tasks: dbt2_setup_user.yml
 
 - name: Include DBT-2 .pgpass configuration
   include_role:
     name: setup_dbt2
     tasks_from: dbt2_configure_pgpass
+
+- name: Make a list of DBT-2 client system addresses
+  set_fact:
+    ssh_targets: "{{ ssh_targets + [ hostvars[item]['private_ip'] ] }}"
+  with_items: "{{ groups['dbt2_client'] }}"
+  when: "{{ 'dbt2_client' in groups }}"
+  register: ssh_targets_list
+
+- name: Make a list of database system addresses
+  set_fact:
+    ssh_targets: "{{ ssh_targets + [ hostvars[item]['private_ip'] ] }}"
+  with_items: "{{ groups['primary'] }}"
+  register: ssh_targets_list
```

Comment:
Not sure you have to use `register`, you can use directly `ssh_targets`:
```yaml
- name: Make a list of DBT-2 client system addresses
  set_fact:
    ssh_targets: "{{ ssh_targets|default([]) + [ hostvars[item]['private_ip'] ] }}"
  with_items: "{{ groups['dbt2_client'] }}"
  when: "{{ 'dbt2_client' in groups }}"

- name: Make a list of database system addresses
  set_fact:
    ssh_targets: "{{ ssh_targets|default([]) + [ hostvars[item]['private_ip'] ] }}"
  with_items: "{{ groups['primary'] }}"

...

- name: Gather SSH public keys from DBT-2 client and database systems.
  ansible.builtin.command: ssh-keyscan {{ item }}
  register: ssh_keyscan
  loop: "{{ ssh_targets }}"
  when: "{{ 'dbt2_client' in groups }}"
```

Reply:
Ok, cleaned that up.